aidies-project/client/src/App.js

```diff
@@ -15,15 +18,17 @@ const PrivateRoute = ({component: Component, ...rest}) => (
   )} />
 )
 const App = () => {  
-  // const [loaded, setLoaded] = useState(false);
-  const needLogin = useSelector(state => !state.authentication.user);
+  const needLogin = useSelector(state => !state.authentication.user.id);
+  console.log(needLogin);
   return (
     <BrowserRouter>
       <Navigation needLogin={needLogin}/>
         <Switch>
-            <Route path="/" component={CampaignBrowser} />
+            <Route exact path="/campaign/create" component={() => <CampaignCreate needLogin={needLogin} />} />
+            <Route exact path="/campaign/:id" component={() => <CampaignDetail needLogin={needLogin} />}  />
```

Comment:
You should refactor this so that `const needLogin = useSelector(state => !state.authentication.user.id);` is used in the `CampaignCreate` and `CampaignDetail` components.
```suggestion
            <Route exact path="/campaign/create" component={CampaignCreate} />
            <Route exact path="/campaign/:id" component={CampaignDetail} />
```

Reply:
You should also do the same thing for `<Navigation needLogin={needLogin}/>`!